.github/workflows/publish_layer.yml

```
@@ -0,0 +1,207 @@
+name: Deploy layer to all regions
+
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  # TODO: remove after setup is finished
+  workflow_dispatch: { }
+  workflow_run:
+    workflows: [ "Publish to PyPi" ]
+    types:
+      - completed
+
+
+jobs:
+  build-layer:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16.12'
+      - name: Setup python
+        uses: actions/setup-python@v4
```

```suggestion
      - name: Setup python
        uses: actions/setup-python@v4
        python-version: '3.9'
        cache: 'pip'
```

Enabling cache and explicit Python version to prevent potential breaks

---

Example with `env` to make it easier to switch to another version later too: https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/workflows/on_merged_pr.yml#L10:L10

---

added cache